src/Akka.Streams.Kafka/Stages/Producers/DefaultProducerStage.cs

```diff
@@ -225,28 +220,58 @@ private void CheckForCompletion()
                 }
             }
         }
-        
-        private void HandleProduceError(IProducer<K, V> producer, Error error)
+
+        private void CloseAndFailStage(Exception ex)
         {
-            Log.Error(error.Reason);
+            CloseProducerImmediately();
+            FailStage(ex);
+        }
+
+        private void CloseProducerImmediately()
```

Comment:
Maybe change the method name if this method might not actually close the producer.

Reply:
Well, it will dispose the producer immediately if it owns the producer. If it is an external producer, we skip the dispose. Difference with the other close method is the other method will attempt to flush the producer first before disposing it.